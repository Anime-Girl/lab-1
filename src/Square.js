import React from 'react'

class Square extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id: props.id,
            CMYK: props.CMYK,
            value: props.value
        };
    }
    render() {
        return (
            <div className="Square">
                <div style={{ backgroundColor: this.state.value }} class="block" id={this.state.id}> {this.state.CMYK} </div>
            </div>
        )
    }
}
export default Square;