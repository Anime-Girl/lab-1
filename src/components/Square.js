import React from 'react'

class Square extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            id: props.id,
            CMYK: props.CMYK,
            value: props.value
        };
    }
    componentWillReceiveProps(newProps) {
        this.setState({
            value: newProps.value,
            CMYK: newProps.CMYK
        })
    }
    render() {
        return (
            <div className="Square">
                <div type="text" class="rectangle" value='this.state.value' style={{ backgroundColor: this.state.value }} id={this.state.id}></div>
                <h1>{this.state.CMYK}</h1>
            </div>
        )
    }
}
export default Square;