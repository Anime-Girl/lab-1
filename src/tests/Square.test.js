import React, { createElement } from 'react';
import { create } from 'react-test-renderer';
import Square from '../components/Square';

// Still makes sense to see if this Component actually renderable
describe('[RENDER] Square â€“ Snapshot test', () => {
  test('Matches the snapshot', () => {
    const square = create(<Square />);
    expect(square.toJSON()).toMatchSnapshot();
  });
});
