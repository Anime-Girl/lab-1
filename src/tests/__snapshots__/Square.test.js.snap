// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Square â€“ Snapshot test Matches the snapshot 1`] = `
<div
  className="Square"
>
  <div
    class="rectangle"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
    type="text"
    value="this.state.value"
  />
  <h1 />
</div>
`;
